app/src/main/java/com/sopt/shinmungo/core/designsystem/component/example/CustomDialog.kt

```diff
@@ -0,0 +1,172 @@
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.sopt.shinmungo.R
+
+@Composable
+fun CustomDialog(
+    title: String,
+    message: String,
+    buttons: List<Pair<String, () -> Unit>>,
+    onIconClick: () -> Unit = {},
+    modifier: Modifier = Modifier
```

Comment:
하나의 컴포저블에서 또다른 컴포저블 함수를 받아서 사용할 수 있어요!
다이얼로그의 경우, 제목과 아이콘을 포함하는 오렌지 색깔의 상단바를 제외한 내부 내용물들이 많이 달라요.

따라서 내부 내용은 외부에서 주입받을 수 있도록 다음과 같이 수정하면 좋을 것 같아요!
```suggestion
@Composable
fun CustomDialog(
    title: String,
    icon: ImageVector
    modifier: Modifier = Modifier,
    onIconClick: () -> Unit = {},
    content: @Composable () -> Unit = {}
```

위와 같이 바꾸게 되면 제목과 아이콘, 그리고 아이콘을 터치하는 이벤트만 이 함수에서 관리하면 되고, content 같은 경우 올바른 위치에 선언만 해주면 됩니다! 이 함수의 `message section`과 `button section`을 `content` 하나로 대체하면 될 것 같아요!